Tweak compatability documentation (#2652)
JuliaLang/Pkg.jl: docs/src/compatibility.md

### Caret specifiers

> Carets allow upgrade that would be compatible according to semver, and are the default behaviour if no specifier is used.

Reviewer: ```suggestion
A caret (`^`) specifier allows upgrade that would be compatible according to semver. This is the default behavior if no specifier is used.
```
